src/main/java/gregtech/common/terminal/app/worldprospector/WorldProspectorARApp.java

```diff
@@ -95,6 +143,20 @@ public boolean mouseClicked(int mouseX, int mouseY, int button) {
                 public void handleClientAction(int id, PacketBuffer buffer) {
                     if (id == -1) {
                         selectReference(index, buttonWidget);
+                    }
+                    else if (id == 0) {
+                        updateBlockSelectionAndColor(ItemStack.EMPTY, index, buttonWidget);
+                    }
+                    else if (id == 1) {
+                        PacketBuffer tmpBuffer = new PacketBuffer(buffer.copy());
+                        try {
+                            ItemStack stack = tmpBuffer.readItemStack();
+                            updateBlockSelectionAndColor(stack, index, buttonWidget);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
```

Comment:
Would we prefer to log an error here, fallback to updating the selection to ItemStack.EMPTY, and then move on?

Reply:
I took that logic from someone else where we read from a buffer.
Will add a logging message.